include/revng/TupleTree/TupleTreeDiff.h

```diff
@@ -514,6 +514,7 @@ TupleTreeDiff<T>::apply(TupleTree<T> &M, revng::ErrorList &EL) const {
       continue;
     }
     tupletreediff::detail::ApplyDiffVisitor<T> ADV{ &C, &EL };
+    M.evictCachedReferences();
```

Comment:
I'm confused, why do this at every iteration of the loop? And why before declaring `ADV` and using it?

Can we do it once at the beginning? Or even better, not do it at all.

I think we could push `evictCachedReferences` into `initializeReferences`.

Reply:
IIRC pushing `evictCachedReferences` exactly before `callByPath` was decided jointly with @pfez, though I actually don't see why it couldn't be hoisted, in principle. It still needs to be done before `callByPath`, I guess. Maybe @pfez can suggest an improvement here?

Reply:
this can be closed